src/control/Control.cpp

```diff
@@ -1923,67 +1923,62 @@ auto Control::newFile(string pageTemplate, fs::path fileName) -> bool {
 /**
  * Check if this is an autosave file, return false in this case and display a user instruction
  */
-auto Control::shouldFileOpen(string filename) -> bool {
-    // Compare case insensitive, just in case (Windows, FAT Filesystem etc.)
+auto Control::shouldFileOpen(fs::path const& file) const -> bool {
+    auto base_path = Util::getConfigSubfolder("");
 
-    filename = StringUtils::toLowerCase(filename);
-    string basename = StringUtils::toLowerCase(Util::getConfigSubfolder("").string());
-
-    if (basename.size() > filename.size()) {
+    if (base_path < file) {
         return true;
     }
 
-    filename = filename.substr(0, basename.size());
-
-    if (filename == basename) {
+    if (file.string().substr(0, base_path.string().size()) == base_path.string()) {
 
         string msg = FS(_F("Do not open Autosave files. They may will be overwritten!\n"
                            "Copy the files to another folder.\n"
                            "Files from Folder {1} cannot be opened.") %
-                        basename);
+                        base_path.string());
         XojMsgBox::showErrorToUser(getGtkWindow(), msg);
         return false;
     }
 
     return true;
```

Comment:
Mhh maybe you create a new function Util::isChild and a test for it and just call this here.

```
#include <filesystem>
#include <iostream>

namespace fs = std::filesystem;

namespace Util{
    bool isChild(fs::path path, fs::path base);
}

bool Util::isChild(fs::path path, fs::path base) {
	auto canonicalize = [](fs::path const& path){
		return fs::exists(path) ? fs::canonical(path) : fs::weakly_canonical(path);
	};

	base = canonicalize(base);
	path = canonicalize(path.remove_filename());

	auto result1 = fs::relative(path, base);
	return *std::begin(result1) !=  "..";
}

int main()
{

	fs::path pr_util("/home/fabian/Bilder/test1");
	fs::path child_util("/home/fabian/Bilder/test1/blah/child.file.txt");
	fs::path non_child("/home/fabian/git/file.txt");


	std::cout << std::boolalpha << Util::isChild(child_util, pr_util) << std::endl;
	std::cout << std::boolalpha << Util::isChild(non_child, pr_util) << std::endl;
}
```